### PR TITLE
dts: arm: st: f1: added DAC

### PR DESCRIPTION
**Pull request by calaghann**
Adds DAC to stm32f103Xc, stm32f105Xc, stm32f107Xc.

> Low-density devices are STM32F101xx, STM32F102xx and STM32F103xx 
> microcontrollers where the Flash memory density ranges between 16 and 32 Kbytes.
> 
> Medium-density devices are STM32F101xx, STM32F102xx and STM32F103xx 
> microcontrollers where the Flash memory density ranges between 64 and 128 Kbytes.
> 
> High-density devices are STM32F101xx and STM32F103xx microcontrollers where the 
> Flash memory density ranges between 256 and 512 Kbytes.
> 
> XL-density devices are STM32F101xx and STM32F103xx microcontrollers where the 
> Flash memory density ranges between 768 Kbytes and 1 Mbyte.
> 
> Connectivity line devices are STM32F105xx and STM32F107xx microcontrollers.
> 
> This section applies to connectivity line, high-density and XL-density STM32F101xx and 
> STM32F103xx devices only.

Only DAC1 is added so far. Tested with stm32f103vc on a custom board.

### CODE REVIEW

File: dts/arm/st/f1/stm32f105Xc.dtsi [hunk 1]
```diff
@@ -18,5 +18,14 @@
...
 		};
+
+		dac1: dac@40007400 {
+			compatible = "st,stm32-dac";
+			reg = <0x40007400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x20000000>;
+			status = "disabled";
+			label = "DAC_1";
+			#io-channel-cells = <1>;
+		};
```
Comment (erwango) [inline dts/arm/st/f1/stm32f105Xc.dtsi:29]: Move this to stm32f105.dtsi, this is still true (all stm32f105 SoCs have a DAC) and will also benefit to stm32f107.dtsi

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for sharing this work.
Would you mind to also add node in stm32f100Xb.dtsi ? This would make the definition complete across series.

Then, one nice addition would be to enable DAC node on au upstream board, so that it could be built on CI. Would you have one? Unfortunately I don't have one that has DAC, so I can't help in this task.

File: dts/arm/st/f1/stm32f105Xc.dtsi (hunk 1 above)
Comment (GeorgeCGV) [reply dts/arm/st/f1/stm32f105Xc.dtsi:29]: @erwango, isn't it the case that DAC is applicable only for

> This section applies to connectivity line, high-density and XL-density STM32F101xx and
STM32F103xx devices only.

and `stm32f100Xb` is considered to be "Medium-density device" (*"ranges between 64 and 128 Kbytes"*)?

Comment (calaghann) [reply dts/arm/st/f1/stm32f105Xc.dtsi]: Added DAC to stm32f100Xb.dtsi and stm32f105.dtsi according to your suggestions

**Review by erwango — APPROVED** [review]

**Review by ABOSTM — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**